### DISCUSSION TITLE
Streamlink --twitch-access token-param where I can find the access token

### DISCUSSION THREAD

**Discussion by bobvargas** [streamlink/streamlink#4962] [Q&A] [answered]
### Streamlink version

Latest build from the master branch

### Description

Hi with the new --twitch-accesstoken-param option where I can find a access token for the twitch front page option because I'm confused with the chrome inspect option thanks.

### Debug log

```text
--
```

**Comment by mkbloke** [comment]
Have you followed the instructions here: https://streamlink.github.io/cli/plugins/twitch.html#authentication, as they seem quite comprehensive?

**Reply by bobvargas** [reply]
I did follow thank you

**Comment by bastimeyer** [comment, marked as answer]
Read this and let me quote and highlight it again:
https://github.com/streamlink/streamlink/issues/4949

> By setting a set of known headers for triggering the generation of different streaming access tokens, e.g. for players where Twitch usually doesn't show ads, like the front page player on their website, embedded web players for specific sites/events, etc. This is explained in the docs linked above. Streamlink had already implemented workarounds several times over the past years by switching to different API call parameters for using a different kind of web player where Twitch doesn't serve ads, and all of those workarounds had to be reverted after a short while again. **It is currently unknown whether any of those web players exist** and which API parameters would be needed. The [--twitch-api-header](https://streamlink.github.io/latest/cli.html#cmdoption-twitch-api-header) and newly added [--twitch-access-token-param](https://streamlink.github.io/latest/cli.html#cmdoption-twitch-access-token-param) CLI parameters allow users to change the API headers and parameters when acquiring a streaming access token, so that if something new gets discovered, users don't have to wait for a new Streamlink release.

The context of that is this and the changelog listed there:
https://github.com/streamlink/streamlink/issues/3210#issue-710685043

----

Twitch's front page player does include ads, so what you're asking is irrelevant.

You're also not looking for THE access token, you're looking how an access token is requested on their GraphQL API. The query for that is `PlaybackAccessToken` (so just filter for network requests with that content in your web browser's dev tools), and one of the variables of that query is `playerType`, which defaults to `embed` in Streamlink's Twitch plugin, which is the value used by, you guessed it, their embeddable web player. The value that gets set by their regular web player is `site`, and the value for their front page web player is `frontpage`.

`--twitch-access-token-param=playerType=frontpage` would therefore be the answer you're looking for. But as said, ads still get embedded into the HLS stream.

**Reply by bobvargas** [reply]
Thanks
